core/src/util/builders.h

```diff
@@ -5,82 +5,72 @@
 #include "tileData.h"
 #include "platform.h"
 
-namespace Builders {
+enum class CapTypes {
+    BUTT = 0, // No points added to end of line
+    SQUARE = 2, // Two points added to make a square extension
+    ROUND = 6 // Six points added in a fan to make a round cap
+};
 
-    /* Build a tesselated polygon
-     * @_polygon input coordinates describing the polygon
-     * @_pointsOut tesselated output coordinates are added to this vector
-     * @_normalsOut normal vectors for each output coordinate are added to this vector
-     * @_indicesOut indices for drawing the polygon as triangles are added to this vector
-     */
-    void buildPolygon(const Polygon& _polygon, std::vector<glm::vec3>& _pointsOut, std::vector<glm::vec3>& _normalsOut, std::vector<int>& _indicesOut);
+enum class JoinTypes {
+    MITER = 0, // No points added at line join
+    BEVEL = 1, // One point added to flatten the corner of a join
+    ROUND = 5 // Five points added in a fan to make a round outer join
+};
+
+struct PolyLineOptions {
+    CapTypes cap;
+    JoinTypes join;
+    float halfWidth;
+};
```

Comment:
Need to have a constructor definition!

A PolyLineOptions is initialized with default values in PolygonStyle::buildLine and since halfWidth has no default value, it will result inconsistent behavior.